src/check/canonicalize/CIR.zig

```diff
@@ -2149,7 +2160,11 @@ pub const Pattern = union(enum) {
     list: struct {
         list_var: TypeVar,
         elem_var: TypeVar,
-        patterns: Pattern.Span,
+        patterns: Pattern.Span, // All non-rest patterns
+        rest_info: ?struct {
+            index: u32, // Where the rest appears (split point)
+            pattern: ?Pattern.Idx, // None for `..`, Some(assign) for `.. as name`
+        },
```

Comment:
This is how I'm representing the spread `..` rest pattern in the list. There can only be one and it can be in any index.